async/kafka/kafka.go

```diff
@@ -39,6 +38,19 @@ func init() {
 	prometheus.MustRegister(topicPartitionOffsetDiff)
 }
 
+// ConsumerConfig is the common configuration of patron kafka consumers
+type ConsumerConfig struct {
+	Brokers      []string
+	Buffer       int
+	DecoderFunc  encoding.DecodeRawFunc
+	SaramaConfig *sarama.Config
+}
+
+//Consumer interface exports the two config objects in order to be modified by option functions.
+type Consumer interface {
+	ConsumerConfig() *ConsumerConfig
```

Comment:
We should not expose the internals here.

Reply:
Same as [here](https://github.com/beatlabs/patron/pull/120/files#r362782963)